multi-buffer-search/src/services/resultFormatter.ts

```
@@ -0,0 +1,163 @@
+          )
+        )
+        for (let i = 0; i < formattedExcerpt.matchLineIndices.length; i++) {
+          const multiLine = startLine + formattedExcerpt.matchLineIndices[i]
+          mapping.lineToExcerpt.set(multiLine, excerpt)
+        }
+        mapping.excerpts.set(excerpt.id, excerpt)
+        const fileKey = uri.toString()
+        if (!mapping.excerptsByFile.has(fileKey)) {
+          mapping.excerptsByFile.set(fileKey, [])
+        }
+        mapping.excerptsByFile.get(fileKey)!.push(excerpt)
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)